rfc/rfc-0000-allow-resources-to-report-status.md

```diff
@@ -0,0 +1,335 @@
+    - name: deployer
+      outputs:
+        ...
+      stampedRef:
+        ...
+      templateRef:
+        apiVersion: carto.run/v1alpha1
+        kind: ClusterTemplate
+        name: deploy
+      stampedStatus:
+        status: 'Unhealthy'
+        fields:
+          - key: status.conditions['ReconcileFailed'].status
+            value: True
+            message: "Some kapp useful error message"
+```
+
+```yaml
+---
+apiVersion: carto.run/v1alpha1
+kind: ClusterResourceStatusRule
+metadata:
+  name: service-rule
+spec:
+  healthy:
+    matchFields:
+      - key: '.status.loadBalancer'
+        operator: 'Exists'
+  unhealthy:
+    matchFields:
+      - key: '.status.loadBalancer'
+        operator: 'DoesNotExist'
+
+---
+apiVersion: carto.run/v1alpha1
+kind: ClusterTemplate
+metadata:
+  name: servicer
+spec:
+  resourceStatusRef:
+    name: service-rule
+
+  template:
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: $(workload.metadata.name)$
+    spec:
+      ...
+
+---
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  ...
+spec:
+  ...
+status:
+  conditions:
+    ...
+  resources:
+    - name: service-creator
+      outputs:
+        ...
+      stampedRef:
+        ...
+      templateRef:
+        apiVersion: carto.run/v1alpha1
+        kind: ClusterTemplate
+        name: servicer
+      stampedStatus:
+        status: 'Healthy'
+        fields:
+          - key: status.loadBalancer
+            value: {}
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+A stamped resource can be in one of three states:
+1. 'Healthy'
+2. 'Unhealthy'
+3. 'Unknown'
+
+It is up to each template author to determine what "success" and "failure" mean for the given resource the template is stamping out. "Unknown" strictly represents that Cartographer has not be able to determine success or failure.
+
+A `ClusterResourceStatusRule` requires defining matchers for both the `Healthy` state and the `Unhealthy` state.
+The matchers in the `Healthy` state are ANDed together, whereas the matchers in the `Unhealthy` state are ORed. That is, a resource is considered to be `Healthy` only if _all of_ the matchers are fulfilled, but is considered to be `Unhealthy` if _any one of_ the matchers are fulfilled.
+
+The supported matchers are `matchConditions` and `matchFields`. Each is an array, and an author can define either or both of these matchers for a given state.
+
+For `matchConditions`, authors need only define the `type` and `status` of the condition for Cartographer to read. In the owner status, the full condition will be reflected (with the `reason`, `message`, etc.)
+For `matchFields`, the spec is based off of `matchFields` in blueprint resource options. Authors need to define the `key` and `operator`, and potentially the `values` (if the operator is `In` or `NotIn`). Authors can also optionally define a `messagePath`, which is a path in the resource where Cartographer can pull off a message to display in the owner status.
```

Comment:
How is the messagePath aggregated when there are multiple matches?

Reply:
I believe this is no longer relevant after the changes made to the RFC.